medusa/server/api/v2/episodes.py

```diff
@@ -0,0 +1,169 @@
+    identifier = ('episode_slug', r'[\w-]+')
+    #: path param
+    path_param = ('path_param', r'\w+')
+    #: allowed HTTP methods
+    allowed_methods = ('GET', 'PATCH', )
+
+    def get(self, series_slug, episode_slug, path_param):
+        """Query episode information.
+
+        :param series_slug: series slug. E.g.: tvdb1234
+        :param episode_number:
+        :param path_param:
+        """
+        series_identifier = SeriesIdentifier.from_slug(series_slug)
+        if not series_identifier:
+            return self._bad_request('Invalid series slug')
+
+        series = Series.find_by_identifier(series_identifier)
+        if not series:
+            return self._not_found('Series not found')
+
+        if not episode_slug:
+            detailed = self._parse_boolean(self.get_argument('detailed', default=False))
+            season = self._parse(self.get_argument('season', None), int)
+            data = [e.to_json(detailed=detailed) for e in series.get_all_episodes(season=season)]
+            return self._paginate(data, sort='airDate')
+
+        episode_number = EpisodeNumber.from_slug(episode_slug)
+        if not episode_number:
+            return self._bad_request('Invalid episode number')
+
+        episode = Episode.find_by_series_and_episode(series, episode_number)
+        if not episode:
+            return self._not_found('Episode not found')
+
+        detailed = self._parse_boolean(self.get_argument('detailed', default=True))
+        data = episode.to_json(detailed=detailed)
+        if path_param:
+            if path_param == 'metadata':
+                data = episode.metadata() if episode.is_location_valid() else {}
+            elif path_param in data:
+                data = data[path_param]
+            else:
+                return self._bad_request("Invalid path parameter '{0}'".format(path_param))
+
+        return self._ok(data=data)
+
+    def patch(self, series_slug, episode_slug=None, path_param=None):
+        """Patch episode."""
+        series_identifier = SeriesIdentifier.from_slug(series_slug)
+        if not series_identifier:
+            return self._bad_request('Invalid series slug')
+
+        series = Series.find_by_identifier(series_identifier)
+        if not series:
+            return self._not_found('Series not found')
+
+        data = json_decode(self.request.body)
+
+        # Multi-patch request
+        if not episode_slug:
+            return self._patch_multi(series, data)
+
+        episode_number = EpisodeNumber.from_slug(episode_slug)
+        if not episode_number:
+            return self._bad_request('Invalid episode number')
+
+        episode = Episode.find_by_series_and_episode(series, episode_number)
+        if not episode:
+            return self._not_found('Episode not found')
+
+        accepted = {}
+        ignored = {}
+        patches = {
+            'status': IntegerField(episode, 'status'),
+            'quality': IntegerField(episode, 'quality'),
```

Comment:
I don't like the fact the patches are declared twice.
Maybe make a `_patch_episode` function that returns true if accepted, false otherwise?